### ISSUE TITLE
false negative for indent rule on class properties

### ISSUE THREAD

**Issue by remcohaszing** [babel/babel-eslint#598]
Static class properties break the indent rule for static class properties, class member fields, and all following class members.

For example, gIven the following `.eslintrc.js`:

```js
module.exports = {
  root: true,
  extends: 'airbnb',
  parser: 'babel-eslint',
  rules: {
    'react/prefer-stateless-function': 'off',
  },
};
```

The following class definition is fine, as expected:

```jsx
import PropTypes from 'prop-types';
import React from 'react';


export default class Foo extends React.Component {
  static propTypes = {
    title: PropTypes.string,
  }

  static defaultProps = {
    title: 'Bar',
  }

  render() {
    const {
      title,
    } = this.props;

    return (
      <h1>
        {title}
      </h1>
    );
  }
}
```

However, the following is (unexpectedly) fine as well:

```jsx
import PropTypes from 'prop-types';
import React from 'react';


export default class Foo extends React.Component {
    static propTypes = {  // indentation doesn’t matter
      title: PropTypes.string,
    }

      static defaultProps = {  // indentation doesn’t matter
        title: 'Bar',
      }

      render() {  // indentation must be aligned with defaultProps.
        const {
          title,
        } = this.props;

        return (
          <h1>
            {title}
          </h1>
        );
      }
}
```

The same goes for class members

```jsx
import PropTypes from 'prop-types';
import React from 'react';


export default class Foo extends React.Component {
      state = {  // indentation doesn’t matter
        title: 'Bar',
      }

      render() {  // indentation must be aligned with state.
        const {
          title,
        } = this.state;

        return (
          <h1>
            {title}
          </h1>
        );
      }
}
```

I would expect the indent rule to warn about indentation of properties as well.

I reported this here, because it’s about stage-X syntax, which is not supported by ESLint natively.

**Comment by 10hendersonm** [comment]
I'm seeing this as well, and haven't found a good solution for it yet.

```
class Create extends Component {
                    state = {
                      clearVisible: false,
                    }
            clearForm = () => {
              this.setState((prevState) => ({
                clearVisible: !prevState.clearVisible,
              }))
            }
  openDialog = () => {
    this.props.setCreateVisibility(true)
  }
  closeDialog = () => {
    if (this.props.create.fields.anomalyId) {
      this.props.clearCreateForm()
    }
    this.props.setCreateVisibility(false)
  }
```

Seems like any class property or arrow function's indentation doesn't matter, and the last one will dictate the indentation level of any regular class function declarations after it.

**Comment by drew-wallace** [comment]
I'm also seeing this.

**Comment by HeleenE** [comment]
Same here, anybody found a solution yet?

**Comment by HQidea** [comment]
same too, and static class properties break the space-infix-ops rule as well

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
